### ISSUE TITLE
Proposed Vault Interface

### ISSUE THREAD

**Issue by mjallday** [verygoodsecurity/starlarky#46]
Used to get or put data from customer's vault. 

Initial implementation would handle only string data, no custom aliasing, no custom ttl

pseudo code as 
```python

class Vault:
  def put(
    val,  # value to store stream of bytes or string
    storage='persistent',  # oneof persistent, volatile
    format='default',  # oneof https://www.verygoodsecurity.com/docs/terminology/nomenclature#alias-formats TODO: for custom alias generators needs to take a function.
    tags=[],
  ):
    # implementation here.
  
  def get(
    alias,
    storage='persistent',  # oneof persistent, volatile.
  )
    # implementation here.
```

**Comment by nestorsokil** [comment]
The change in https://github.com/verygoodsecurity/operation-pipeline/pull/423 will also support transparent batch put/get. The users can just supply a list of values and receive a list of results.
```python
tokens = vault.put(['secret_1', 'secret_2'], storage='persistent')
for tok in tokens:
    do_stuff(tok)
secrets = vault.get(tokens)
```

**Comment by mahmoudimus** [comment]
@nestorsokil is there a PR coming for a larky lib here?

**Comment by mjallday** [comment]
implementing via https://github.com/verygoodsecurity/starlarky/pull/109

**Comment by moehajj** [comment]
merged in #109
